pkg/cloud-init/cloud-init_test.go

```diff
@@ -195,254 +217,391 @@ var _ = Describe("CloudInit", func() {
+							virtClient.EXPECT().CoreV1().Return(networkClient.CoreV1()),
+						)
+
+						cloudInitData := &v1.CloudInitNoCloudSource{
+							UserDataSecretRef:    &k8sv1.LocalObjectReference{Name: userDataSecretName},
+							NetworkDataSecretRef: &k8sv1.LocalObjectReference{Name: networkDataSecretName},
+						}
 
-					userSecret := &k8sv1.Secret{
-						ObjectMeta: metav1.ObjectMeta{
-							Name:      "userDataSecretName",
-							Namespace: namespace,
-						},
-						Type: "Opaque",
-						Data: map[string][]byte{
-							"userdata": []byte("secretUserData"),
-						},
-					}
-					userClient := fake.NewSimpleClientset(userSecret)
-
-					networkSecret := &k8sv1.Secret{
-						ObjectMeta: metav1.ObjectMeta{
-							Name:      "networkDataSecretName",
-							Namespace: namespace,
-						},
-						Type: "Opaque",
-						Data: map[string][]byte{
-							"networkdata": []byte("secretNetworkData"),
-						},
-					}
-					networkClient := fake.NewSimpleClientset(networkSecret)
+						err := resolveNoCloudSecrets(cloudInitData, namespace, virtClient)
+						Expect(err).To(BeNil())
+						Expect(cloudInitData.UserData).To(Equal("secretUserData"))
+						Expect(cloudInitData.NetworkData).To(Equal("secretNetworkData"))
+					})
+
+					It("should succeed to verify nothing", func() {
+						fakeClient := fake.NewSimpleClientset()
+						virtClient.EXPECT().CoreV1().Return(fakeClient.CoreV1())
+						cloudInitData := &v1.CloudInitNoCloudSource{}
+						err := resolveNoCloudSecrets(cloudInitData, namespace, virtClient)
+						Expect(err).To(BeNil())
+					})
+
+					It("should fail to verify UserDataSecretRef without a secret", func() {
+						fakeClient := fake.NewSimpleClientset()
+						virtClient.EXPECT().CoreV1().Return(fakeClient.CoreV1())
+
+						cloudInitData := &v1.CloudInitNoCloudSource{
+							UserDataSecretRef: &k8sv1.LocalObjectReference{Name: userDataSecretName},
+						}
 
-					gomock.InOrder(
-						virtClient.EXPECT().CoreV1().Return(userClient.CoreV1()),
-						virtClient.EXPECT().CoreV1().Return(networkClient.CoreV1()),
-					)
+						err := resolveNoCloudSecrets(cloudInitData, namespace, virtClient)
+						Expect(err.Error()).To(Equal(fmt.Sprintf("secrets \"%s\" not found", userDataSecretName)))
+					})
 
-					cloudInitData := &v1.CloudInitNoCloudSource{
-						UserDataSecretRef:    &k8sv1.LocalObjectReference{Name: "userDataSecretName"},
-						NetworkDataSecretRef: &k8sv1.LocalObjectReference{Name: "networkDataSecretName"},
-					}
+					It("should fail to verify NetworkDataSecretRef without a secret", func() {
+						fakeClient := fake.NewSimpleClientset()
+						virtClient.EXPECT().CoreV1().Return(fakeClient.CoreV1())
 
-					err := ResolveSecrets(cloudInitData, namespace, virtClient)
-					Expect(err).To(BeNil())
-					Expect(cloudInitData.UserData).To(Equal("secretUserData"))
-					Expect(cloudInitData.NetworkData).To(Equal("secretNetworkData"))
-				})
-			})
-			Context("with nothing", func() {
```

Comment:
does this test still exist, or was it removed?

Reply:
The test still exists, it is now named "should succeed to verify nothing". See https://github.com/kubevirt/kubevirt/pull/2120/files/e61e2644a86e6f96d175bd8714de117a9c550cc8#diff-03390e87555cbbbe9b1ee8ab733f678eR354